src/resolve/resolved-dns-packet.c

```diff
@@ -1837,6 +1837,9 @@ int dns_packet_read_rr(DnsPacket *p, DnsResourceRecord **ret, bool *ret_cache_fl
                 if (r < 0)
                         return r;
 
+                if (rdlength - 4 > rdlength)
+                        return -EOVERFLOW;
```

Comment:
I'd really treat this as -EBADMSG here, after all the caller just wants to have this parsed and an overflow is definitely one ind of invalid msg